kartothek/core/factory.py

```diff
@@ -39,12 +45,12 @@ class DatasetFactory(DatasetMetadataBase):
 
     def __init__(
         self,
-        dataset_uuid,
-        store_factory,
-        load_schema=True,
-        load_all_indices=False,
-        load_dataset_metadata=True,
-    ):
+        dataset_uuid: str,
+        store_factory: Callable[[], "KeyValueStore"],
```

Comment:
`store_factory()` is called a bit down, wouldn't this fail for `KeyValueStore`?

Reply:
We call `_check_callable` which should fail for just `KeyValueStore`.

Reply:
my bad, didn't read the types properly